packages/paste-core/components/form/src/select/Select.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import css from '@styled-system/css';
+import {Box} from '@twilio-paste/box';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+import {Prefix} from '../shared/Prefix';
+import {Suffix} from '../shared/Suffix';
+import {FieldWrapper} from '../shared/FieldWrapper';
+import {safelySpreadFormControlProps} from '../shared/Utils';
+
+export interface SelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
+  id: string;
+  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  insertBefore?: React.ReactNode;
+  insertAfter?: React.ReactNode;
+  hasError?: boolean;
+  value: string | string[];
+}
+
+export interface SelectIconWrapperProps {
+  ref?: string;
+  insertAfter?: React.ReactNode;
+  children: React.ReactNode;
+}
+
+const SelectIconWrapper = React.forwardRef<HTMLSelectElement, SelectIconWrapperProps>(
+  ({children, insertAfter, ...props}, ref) => (
+    <Box
+      paddingTop="space10"
+      position="absolute"
+      display="inline-flex"
+      alignItems="center"
+      pointerEvents="none"
+      top="space120"
+      right={insertAfter ? 'space160' : 'space60'}
+      zIndex="zIndex10"
+      ref={ref}
+      {...props}
+    >
+      {children}
+    </Box>
+  )
+);
+
+const SelectElement = styled.select(
+  css({
+    appearance: 'none',
+    border: 'none',
+    background: 'transparent',
+    outline: 'none',
+    resize: 'none',
+    display: 'block',
+    width: '100%',
+    fontFamily: 'inherit',
+    fontSize: 'fontSize30',
+    lineHeight: 'lineHeight30',
+    fontWeight: 'fontWeightNormal',
+    color: 'colorText',
+    paddingTop: 'space30',
+    paddingRight: 'space40',
+    paddingBottom: 'space30',
+    paddingLeft: 'space40',
+    borderRadius: 'borderRadius20',
+    boxShadow: 'none',
+
+    '&:disabled': {
+      cursor: 'not-allowed',
+      color: 'colorTextWeak',
+    },
+  })
+);
+
+const Select = React.forwardRef<HTMLSelectElement, SelectProps>(
+  ({disabled, hasError, insertBefore, insertAfter, children, size, multiple, ...props}, ref) => (
+    <FieldWrapper disabled={disabled} hasError={hasError}>
+      {insertBefore && <Prefix>{insertBefore}</Prefix>}
+      <Box display="flex" width="100%">
```

Comment:
@richbachman added this to the `<Select />` component, is this what you were thinking, or would you prefer it be another separate component?

Reply:
If this wrapper is relatively positioned, you shouldn't need to know whether or not the insertAfter is present. The absolute position is always based on this Box.

Reply:
@SiTaggart I'm using absolute positioning here because I noticed that if I used relative, the click event wasn't propagating correctly. In the commit I'm about to push I added normalizing spacing for the "right" value with absolute positioning, and gave the icon size 20 padding on the right. Is that an acceptable work around?

Reply:
Then you do a little 

```
top: 50%
transform: translateY(-50%)
```
on the icon wrapper and it's vertically centered

Reply:
Yeah I noticed that in the Chakra implementation, so then I can leave this as `position: absolute` and it's preferable to use the above styles?

Reply:
I'm guessing the use of a computed `transform` here is more reliable than hard-coding the right spacing value?

Reply:
No, the icon wrapper is absolute to the `Box`, relatively positioned wrapper. Then you can get rid of the top padding on the icon wrapper. 

Also curious why you are using right padding and right positioning together? Can it just be right positioning?

Also from what I can see the pointer events still passes through to the underlying select element in FF and chrome locally.